diff --git a/fs/exec.c b/fs/exec.c
index 08f3e4e..5405479 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1713,6 +1713,12 @@ static int format_corename(struct core_name *cn, long signr)
 				err = cn_printf(cn, "%d",
 					      task_tgid_vnr(current));
 				break;
+			case 'i':
+				err = cn_printf(cn, "%d", task_pid_vnr(current));
+				break;
+			case 'I':
+				err = cn_printf(cn, "%d", task_pid_nr(current));
+				break;
 			/* uid */
 			case 'u':
 				err = cn_printf(cn, "%d", cred->uid);
